### PR TITLE
fix error handling in sendRawTransaction method

### PR DESCRIPTION
**Pull request by Flouse**
**Test case:**
```sh
curl -v http://127.0.0.1:8024 -X POST -H "Content-Type: application/json" -d '{
"jsonrpc":"2.0",
"method":"eth_sendRawTransaction",
"params":["0xf901e980017b8080b9019e608060405234801561001057600080fd5b5061017e806100206000396000f3fe608060405234801561001057600080fd5b50600436106100365760003560e01c80632e64cec11461003b5780636057361d14610059575b600080fd5b610043610075565b604051610050919061010c565b60405180910390f35b610073600480360381019061006e91906100d4565b61007e565b005b60008054905090565b806000819055507f93fe6d397c74fdf1402a8b72e47b68512f0510d7b98a4bc4cbdf6ac7108b3c59816040516100b4919061010c565b60405180910390a150565b6000813590506100ce81610131565b92915050565b6000602082840312156100e657600080fd5b60006100f4848285016100bf565b91505092915050565b61010681610127565b82525050565b600060208201905061012160008301846100fd565b92915050565b6000819050919050565b61013a81610127565b811461014557600080fd5b5056fea2646970667358221220153abfb1aecc059411aeeb56035a2223c4fe356d2c82a548b13afc2150dcf41f64736f6c634300080300332aa0d9d55da17b784a014415a47a4459f0aaa131e0f78694315c522214663b43762ba05e1c0ce862f989264f97b855dc1e592bb2f952e5848eaa80c429cd52434eef4e"],
"id":49}'
```

```log
[1] Error: l2_transaction.raw is missing from_id!
[1]     at normalizeObject (/godwoken-web3/packages/godwoken/lib/types.js:46:19)
[1]     at NormalizeRawL2Transaction (/godwoken-web3/packages/godwoken/lib/types.js:60:12)
[1]     at /godwoken-web3/packages/godwoken/lib/types.js:54:16
[1]     at normalizeObject (/godwoken-web3/packages/godwoken/lib/types.js:48:23)
[1]     at Object.NormalizeL2Transaction (/godwoken-web3/packages/godwoken/lib/types.js:69:12)
[1]     at Eth.sendRawTransaction (/godwoken-web3/packages/api-server/lib/methods/modules/eth.js:638:115)
[1]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

Respone after bug fixed:
```JSON
{"jsonrpc":"2.0","id":49,"error":{"code":-32602,"message":"l2_transaction.raw is missing from_id!"}
```

### CODE REVIEW

File: packages/api-server/src/methods/validator.ts [hunk 1]
```diff
@@ -20,6 +20,8 @@ export function middleware(
     params: any[] = [],
     cb: (err: any, val?: any) => void
   ) {
+    console.log("[validator middleware] -", params);
```
Comment (jjyr) [inline packages/api-server/src/methods/validator.ts:23]: We should remove this in the production environment. The content of a transaction may be very large.

File: packages/api-server/src/methods/modules/eth.ts [hunk 2]
```diff
@@ -814,14 +816,23 @@ export class Eth {
...
+        schemas.SerializeL2Transaction(types.NormalizeL2Transaction(rawTx))
+      ).serializeJson();
+      const result = await this.rpc.submit_l2transaction(moleculeTx);
+      console.log('sendRawTransaction hash:', result);
+      callback(null, result);
+    } catch (error) {
+      console.error(error);
+      callback({
+        // https://www.jsonrpc.org/specification | 5.1 Error object
+        code: INTERNAL_ERROR,
```
Comment (jjyr) [inline packages/api-server/src/methods/modules/eth.ts:832]: This is not an internal error, users may submit invalid data, and these behaviors are in our expectation, IMO we should return an `INVALID_PARAMS` error code.

File: packages/api-server/src/methods/validator.ts (hunk 1 above)
Comment (Flouse) [reply packages/api-server/src/methods/validator.ts]: https://github.com/nervosnetwork/godwoken-web3/pull/4/commits/4c6f5e3746b39bd4d8f5181c3ed906164aeb7b45

File: packages/api-server/src/methods/modules/eth.ts (hunk 2 above)
Comment (Flouse) [reply packages/api-server/src/methods/modules/eth.ts]: hupeng has designed a [middleware for parameters validation,](https://github.com/bitrocks/godwoken-web3/blob/4c6f5e3746b39bd4d8f5181c3ed906164aeb7b45/packages/api-server/src/methods/validator.ts#L14-L48)
handling `INVALID_PARAMS` errors.

But this sendRawTransaction method was not using the parameters validation. 
I am not sure why we didn't use it, so I choose `INTERNAL_ERROR` error temporally and make [a little optimization](https://github.com/nervosnetwork/godwoken-web3/pull/4/commits/3fe01b0ade6f77eaf5bc8901c3ba075657e625a9).

**Review by jjyr — APPROVED** [review]

**Pull request merged**